### ISSUE TITLE
Should report CS8600 rather than CS8606 assigning to an iteration variable

### ISSUE THREAD

**Issue by cston** [dotnet/roslyn#40164]
Non-safety nullable warnings should use a single warning (CS8600) to allow those warnings to be suppressed easily (`#pragma warning disable 8600`). Currently, assigning to an iteration variable uses a distinct warning.
```C#
#nullable enable
class Program
{
    static void F(object?[] x)
    {
         foreach (object y in x)
         {
         }
    }
}
```
```
(6,26): warning CS8606: Possible null reference assignment to iteration variable
```
